python/lsst/ip/isr/measureCrosstalk.py

```diff
@@ -0,0 +1,240 @@
+#
+# LSST Data Management System
+# Copyright 2008-2017 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+"""
+Measure intra-CCD crosstalk coefficients.
+"""
+from __future__ import absolute_import, division, print_function
+
+__all__ = ["extractCrosstalkRatios", "measureCrosstalkCoefficients",
+           "MeasureCrosstalkConfig", "MeasureCrosstalkTask"]
+
+from builtins import range
+
+import itertools
+import numpy as np
+
+from lsst.afw.detection import FootprintSet, Threshold
+from lsst.pex.config import Config, Field, ListField, ConfigurableField
+from lsst.pipe.base import CmdLineTask
+
+from .crosstalk import calculateBackground, extractAmp
+from .isrTask import IsrTask
+
+
+def extractCrosstalkRatios(exposure, threshold=30000, badPixels=["SAT", "BAD", "INTRP"]):
+    """Extract crosstalk ratios between different amplifiers
+
+    For pixels above ``threshold``, we calculate the ratio between each
+    target amp and source amp. We return a list of ratios for each pixel
+    for each target/source combination, as a matrix of lists.
+
+    Parameters
+    ----------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure for which to measure crosstalk.
+    threshold : `float`
+        Lower limit on pixels for which we measure crosstalk.
+    badPixels : `list` of `str`
+        Mask planes indicating a pixel is bad.
+
+    Returns
+    -------
+    ratios : `list` of `list` of `list` of `float`
```

Comment:
I can't think of a better description. As mentioned in the docstring, it's a matrix of lists of floats.